From 9cd8865903fbd150d38a46cc21ab3a307541c7fe Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Wed, 1 Nov 2023 13:46:54 +0800
Subject: [PATCH 2/9] SM1&G12B: cold start time optimization. [3/3]

PD#SWPL-139535

Problem:
Cold start time optimization.

Solution:
1. Modify emmc clock mode to HS400.

Verify:
local PB

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: I1a1b0eb7210118ca5875447409ea62600fe4f140
---
 arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts  | 4 ++--
 arch/arm64/boot/dts/amlogic/mesong12b.dtsi              | 2 +-
 arch/arm64/boot/dts/amlogic/mesonsm1.dtsi               | 2 +-
 arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts | 4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
index bcb8dcbb4..e94b81155 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
@@ -1424,8 +1424,8 @@
 	non-removable;
 	disable-wp;
 	//mmc-ddr-1_8v;
-	//mmc-hs200-1_8v;
-	//mmc-hs400-1_8v;
+	mmc-hs200-1_8v;
+	mmc-hs400-1_8v;
 	//mmc-pwrseq = <&emmc_pwrseq>;
 	//vmmc-supply = <&vcc_3v3>;
 	//vqmmc-supply = <&vddio_boot>;
diff --git a/arch/arm64/boot/dts/amlogic/mesong12b.dtsi b/arch/arm64/boot/dts/amlogic/mesong12b.dtsi
index 3c12ea2b4..44a48ada1 100644
--- a/arch/arm64/boot/dts/amlogic/mesong12b.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesong12b.dtsi
@@ -1557,7 +1557,7 @@
 				<&clkc CLKID_FCLK_DIV2>,
 				<&clkc CLKID_FCLK_DIV2P5>;
 		clock-names = "core","mux0","mux1","clkin0","clkin1","clkin2";
-		tx_delay = <0>;
+		tx_delay = <16>;
 		card_type = <1>;
 		mmc_debug_flag;
 		disable-wp;
diff --git a/arch/arm64/boot/dts/amlogic/mesonsm1.dtsi b/arch/arm64/boot/dts/amlogic/mesonsm1.dtsi
index 9d57c32ec..b1047c69b 100644
--- a/arch/arm64/boot/dts/amlogic/mesonsm1.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesonsm1.dtsi
@@ -1750,7 +1750,7 @@
 				<&clkc CLKID_FCLK_DIV2>,
 				<&clkc CLKID_FCLK_DIV2P5>;
 		clock-names = "core","mux0","mux1","clkin0","clkin1","clkin2";
-		tx_delay = <0>;
+		tx_delay = <16>;
 		card_type = <1>;
 		mmc_debug_flag;
 		disable-wp;
diff --git a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
index e2eeddf25..c7e06b04c 100644
--- a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
@@ -1915,8 +1915,8 @@
 	non-removable;
 	disable-wp;
 	//mmc-ddr-1_8v;
-	//mmc-hs200-1_8v;
-	//mmc-hs400-1_8v;
+	mmc-hs200-1_8v;
+	mmc-hs400-1_8v;
 	//mmc-pwrseq = <&emmc_pwrseq>;
 	//vmmc-supply = <&vcc_3v3>;
 	//vqmmc-supply = <&vddio_boot>;
-- 
2.25.1

